### DISCUSSION TITLE
use asyncio.Runner(loop_factory=...) instead of the policy system on 3.11

### DISCUSSION THREAD

**Discussion by graingert** [Kludex/uvicorn#1521] [Ideas]
As part of the GELectomy (removing the per thread global eventloop from `asyncio.get_event_loop()` ) asyncio.runners now provides a policy free way to select the eventloop implementation. uvicorn should use this on 3.11 and consider backporting the implementation for other python versions

**Comment by Kludex** [comment]
Sure. Let's wait for the 3.11 support on underneath packages and go for it. 🙏

**Comment by Kludex** [comment]
Back here 👋 

So...

> It might be better to copy the implementation from python 3.10 asyncio.runners and ignore the broken implementation in 3.11

But shouldn't we wait for things to be fixed in CPython? 🤔
